subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/ntlm/NTLMCredentials.java

```diff
@@ -32,6 +32,11 @@
     public NTLMCredentials(PasswordCredentials credentials) {
         String domain;
         String username = credentials.getUsername();
+
+        if (username == null) {
+            throw new IllegalArgumentException("Username must not be null!");
+        }
+        
```

Comment:
On second thought, please use `Preconditions.checkNotNull` which you can use like this:


```java
String username = Preconditions.checkNotNull(credentials.getUsername(), "username must not be null");
```